prometheus/collectors/dbstats_collector_go115.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2021 The Prometheus Authors
```

Comment:
I would call this file `dbstats_collector_pre_1.15.go`.

Also, I think, you can avoid duplicating most of the code here.

For example, you could have the `Desc` for `MaxIdleTimeClosed` in any case, but do not `Describe` it and do not `Collect` the corresponding metric. Instead, have additional method calls there `DescribeGo115Stats` and `CollectGo115Stats`, and then just have those methods here (as empty), and the proper methods in a separate file with `// +build go1.15` as a tag.